src/utils/send-request.js

```diff
@@ -13,9 +13,21 @@ const log = require('debug')('ipfs-http-client:request')
 
 // -- Internal
 
-function parseError (res, cb) {
+function parseError (res, cb, isJson = true) {
```

Comment:
`isJson` should be required and passed before `cb` - it's very unusual for a function that takes a callback for it to be passed in any position other than the last.

```suggestion
function parseError (res, isJson, cb) {
```

Reply:
Changing the parameter positions would be a breaking change. Is that ok? Is there no where else this `parseError` method being called currently? We may have to change it in all places.

If we look into it, this `isJson` value is actually being computed from the `res`, which itself is being passed as first parameter - so I have no objection to *deduce* the `isJson` again from the `res` inside the `parseError`. That way we would not need to pass the `isJson` as additional parameter (and also avoids the breaking change).

Now, to avoid the code-duplication of computing the `isJson` from `res`, I can extract it into a new method, something like:
````
function isJson(res) {
    return res.headers['content-type'] && res.headers['content-type'].indexOf('application/json') === 0;
}
function parseError(res, cb) {
    if(!isJson(res)) return streamToValue(...);
    streamToJsonValue(...);
}
````
This puts the burden of checking the data on the method itself, rather than on the caller (who should not worry about how the `parseError` does its job).

Though, I initially had the temptation to *fix* the `streamToJsonValue` itself to make it foolproof (instead of modifying `parseError`), since `streamToJsonValue` is internally calling `streamToValue` anyway. After all, `parseError` may not be the only client the `streamToJsonValue` may have, and every one cannot keep checking which variation of `streamToJsonValue` or `streamToValue` should be called in every place. Instead, the `streamToJsonValue` calls `JSON.parse(data)` only for what it believes as JSON data, and behaves like `streamToValue` in all other, non-JSON cases. For example, the below code works perfectly fine without having to modify `parseError` or any other existing `streamToJsonValue` calls:
````
function streamToJsonValue (res, cb) {
  streamToValue(res, (err, data) => {

   ...

    let res = data    // <- proposed change
    try {
      res = JSON.parse(data)
    } catch (err) {
      ; // <- proposed change, no error callback
    }

    cb(null, res)  //<- this will send `data` itself as the result
  })
}
````
This should not break any existing *working* code + more resilient. The `parseError` implementation need not be touched at all (i.e. no need for `isJson` checks, since `streamToJsonValue` takes care of it), though not sure how compatible this behavior is with overall design.

Let me know which approach should be taken forward, will modify accordingly.

Reply:
I'm happy for `isJson` to be extracted into ta function and be called again from `parseError` 👍

`parseError` isn't exported from this module AFAIK so this is the only place that is using it.

Please leave `streamToJsonValue` as it is in this PR - IMHO it's unexpected for it to silently ignore a parse error and return a value that hasn't been parsed as JSON.

Reply:
Done. The local variable `isJson` was clashing with the function name, so named the method as `hasJSONHeaders`.

Reverted back the `parseError` method signature to its earlier form. Kept the conditional routing to `streamToJsonValue`